lib/off_broadway/redis/redix_client.ex

```diff
@@ -16,30 +16,27 @@ defmodule OffBroadway.Redis.RedixClient do
   @default_max_number_of_messages 10
   @max_num_messages_allowed 20
 
+  defstruct [:redis_instance, :list_name, :working_list_name, :receive_messages_opts, :config]
+
   @impl true
   def init(opts) do
     with {:ok, redis_instance} <- validate(opts, :redis_instance),
          {:ok, list_name} <- validate(opts, :list_name),
          {:ok, working_list_name} <- validate(opts, :working_list_name),
          {:ok, receive_messages_opts} <- validate_receive_messages_opts(opts),
          {:ok, config} <- validate(opts, :config, []) do
-      ack_ref =
-        Broadway.TermStorage.put(%{
-          redis_instance: redis_instance,
-          list_name: list_name,
-          working_list_name: working_list_name,
-          config: config
-        })
-
-      {:ok,
-       %{
-         redis_instance: redis_instance,
-         list_name: list_name,
-         working_list_name: working_list_name,
-         receive_messages_opts: receive_messages_opts,
-         config: config,
-         ack_ref: ack_ref
-       }}
+      state = %__MODULE__{
+        redis_instance: redis_instance,
+        list_name: list_name,
+        working_list_name: working_list_name,
+        receive_messages_opts: receive_messages_opts,
+        config: config
+      }
+
+      ack_ref = make_ref()
+      put_config(ack_ref, state)
+
+      {:ok, ack_ref}
```

Comment:
I can't comment there but I think there is a bug on line 44 as receive_messages/2 was not updated to match your changes in init/1

Updating the function like this works for me:

```  @impl true
  def receive_messages(demand, ack_ref) do
    opts = get_config(ack_ref)
    receive_messages_opts = put_max_number_of_items(opts.receive_messages_opts, demand)

    opts.redis_instance
    |> pop_messages(
      opts.list_name,
      opts.working_list_name,
      receive_messages_opts[:max_number_of_items]
    )
    |> wrap_received_messages(ack_ref)
  end

Reply:
Tests pass either way because this file is not tested, but I can't help you there because I'm not competent in writing elixir tests (yet) ;)